rollback/types.go

```diff
@@ -0,0 +1,64 @@
+package rollback
+
+import (
+	"github.com/metrumresearchgroup/pkgr/cran"
+	"github.com/metrumresearchgroup/pkgr/desc"
+	"github.com/metrumresearchgroup/pkgr/gpsr"
+	"github.com/metrumresearchgroup/pkgr/pacman"
+	"github.com/spf13/afero"
+	"github.com/thoas/go-funk"
+)
+
+type RollbackPlan struct {
+	AllPackages []string
+	NewPackages []string
+	UpdateRollbacks []pacman.UpdateAttempt
+	InstallPlan gpsr.InstallPlan
+	Library string
+}
+
+//Constructor
+func CreateRollbackPlan (library string, installPlan gpsr.InstallPlan, preinstalledPackages map[string]desc.Desc) RollbackPlan {
+	ap := installPlan.GetAllPackages()
+	np := DiscernNewPackages(installPlan.GetAllPackages(), preinstalledPackages)
+	return RollbackPlan {
+			AllPackages: ap,
+			NewPackages: np,
+			InstallPlan: installPlan,
+			Library: library,
+		}
+}
+
+
+func (rp *RollbackPlan) PreparePackagesForUpdate(fs afero.Fs, library string) {
+
+	//InstallPlan is aware of _all_ preinstalled packages, even if they're not in pkgr.yml. We don't want to touch
+	//preinstalled packages that weren't in pkgr.yml, so we just want to take the active packages from the install plan.
+	outdatedPackages := rp.InstallPlan.OutdatedPackages
+	var opFiltered []cran.OutdatedPackage
+	for _, op := range outdatedPackages {
+		if funk.Contains(rp.AllPackages, op.Package) {
+			opFiltered = append(opFiltered, op)
+		}
+	}
+
+	// Wrap this function for now until we're ready to move it over.
+	updateAttempts := pacman.PreparePackagesForUpdate(fs, library, opFiltered)
+	rp.UpdateRollbacks = updateAttempts
+}
+
+func DiscernNewPackages(toInstallPackageNames []string, preinstalledPackages map[string]desc.Desc) []string {
+	newPackages := make([]string, 0)
```

Comment:
for consistency generally initialize a new variable with empty length (the default) by just declaring it

var newPackages []string

Reply:
Sure thing. Funnily enough, I was actually doing it _this_ way for consistency -- we don't use var declarations much, and so I was aiming to avoid using them (I remember reading a recommendation somewhere that said aim to use var declarations XOR := operators, but I agree with your recommendation here)